### ISSUE TITLE
slow response

### ISSUE THREAD

**Issue by franperezlopez** [careteditor/caret#582]
Hi,

I'm testing caret, using windows 10 (laptop i7, 16Gb RAM and SSD). I've got a folder which is a GITHUB repo, which contains several markdown files and images. every time that I double click in some file to open it up, it takes like 5 seconds. is this the normal response time?? In every other aspect, I really like this app, but I can't accept taking so much to open up a file

kind regards,

**Comment by astoilkov** [comment]
Hi @franperezlopez,

We have improved Caret app performance. The improvement will be available in the next Caret 4.0 release.

Is it possible you also send us some slowly opening file so we could test it on our end to ensure everything is working correctly in the new version?

Thanks in advance.

**Comment by franperezlopez** [comment]
I didn't realize of the bottom bar ... there is a error there:


The following error has occurred:
```
{
    "message": "Could not call remote function ''. Check that the function signature is correct. Underlying error: Error processing argument at index 0, conversion failure from null\nError: Could not call remote function ''. Check that the function signature is correct. Underlying error: Error processing argument at index 0, conversion failure from null\n    at callFunction (C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\browser\\rpc-server.js:257:11)\n    at EventEmitter.<anonymous> (C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\browser\\rpc-server.js:357:5)\n    at emitMany (events.js:127:13)\n    at EventEmitter.emit (events.js:204:7)\n    at WebContents.<anonymous> (C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\browser\\api\\web-contents.js:256:13)\n    at emitTwo (events.js:106:13)\n    at WebContents.emit (events.js:194:7)",
    "version": "3.4.6",
    "os": "Windows 10",
    "path": "C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\renderer\\api\\remote.js",
    "code": "throw new Error(meta.message + '\\n' + meta.stack)",
    "line": 234,
    "column": 7,
    "stack": [
        "Error: Could not call remote function ''. Check that the function signature is correct. Underlying error: Error processing argument at index 0, conversion failure from null",
        "at callFunction ((C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\browser\\rpc-server.js:257:11)",
        "at EventEmitter.<anonymous> ((C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\browser\\rpc-server.js:357:5)",
        "at emitMany ((events.js:127:13)",
        "at EventEmitter.emit ((events.js:204:7)",
        "at WebContents.<anonymous> ((C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\browser\\api\\web-contents.js:256:13)",
        "at emitTwo ((events.js:106:13)",
        "at WebContents.emit ((events.js:194:7)",
        "at metaToValue ((C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\renderer\\api\\remote.js:234:13)",
        "at Object.remoteMemberFunction ((C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\electron.asar\\renderer\\api\\remote.js:118:18)",
        "at e.value ((C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\app.asar\\extensions\\Menu\\Menu.js:5:2108)",
        "at t.value ((C:\\Users\\fran\\AppData\\Local\\Caret\\app-3.4.6\\resources\\app.asar\\extensions\\TitleBar.js:5:3477)"
    ]
}
```

do you still need the files?? then, let me contact you via email

fran

**Comment by astoilkov** [comment]
If it is possible. Yes.

The error is associated with the menu opening in the top left corner, not with the file performance.

**Comment by astoilkov** [comment]
You could send the file via email at astoilkov@caret.io

**Comment by franperezlopez** [comment]
... any update on the issue??

**Comment by astoilkov** [comment]
It seems this issue is related to reports we have been getting on Caret slowly opening files. You could learn more about it [here](https://github.com/careteditor/caret/issues/539). If you read the discussion you will understand that we haven't been able to reproduce the issue but we hope the next release will fix it because we made a lot of performance improvements.

I hope this information helps.

**Comment by franperezlopez** [comment]
any ETA for the new beta? can it be downloaded somewhere??

**Comment by astoilkov** [comment]
We will release Caret 4.0 Beta on 20 December which is Wednesday next week.

Edit: I have moved the date one day.

**Comment by franperezlopez** [comment]
Hi there ... where can I download the beta ??

**Comment by erusev** [comment]
Hi, we've postponed the release date for the beta to the 3rd of January. We'll tweet and email about it when we release it. We'll also add a link on the website.

**Comment by astoilkov** [comment]
Hi @franperezlopez,

I am not sure if the information got to you but we released [Caret 4.0 Beta](https://caret.io/4beta/) that includes the performance improvements.

Closing this issue. Let us know if you have any feedback.
